src/dbt_client/dbtCoreIntegration.ts

```diff
@@ -223,17 +215,19 @@ export class DBTCoreProjectIntegration
 
   private async createPythonDbtProject() {
     await this.python.ex`from dbt_integration import *`;
+    const profilesDir = await this.findProfilesDirectory();
+    const targetPath = setEnvVariableValue("DBT_TARGET_PATH");
     await this.python
-      .ex`project = DbtProject(project_dir=${this.projectRoot.fsPath}, profiles_dir=${this.dbtProfilesDir}) if 'project' not in locals() else project`;
+      .ex`project = DbtProject(project_dir=${this.projectRoot.fsPath}, profiles_dir=${profilesDir}, target_path=${targetPath}) if 'project' not in locals() else project`;
+    this.disposables.push(
+      // when the project config changes we need to re-init the dbt project
+      ...setupWatcherHandler(this.dbtProfileWatcher, () =>
+        this.rebuildManifest(),
+      ),
```

Comment:
This will create watcher handlers every time the project is refreshed.

Reply:
moved this to initializer